accounting/invoice/utils.py

```diff
@@ -85,22 +106,27 @@ def send_email_with_invoice(template, subject, extra_email_context=None, client_
     template = get_template(template)
     for client_username in client_usernames:
         client = Account.objects.get(user__username=client_username)
-        providers = [rate.provider for rate in client.client_hourly_rates.all()]
+        providers = [rate.provider for rate in client.client_hourly_rates.all() if rate.active]
         for provider in providers:
-            # Copy over certain details from the latest invoice from the proper month to the new invoice.
-            invoice = models.Invoice.objects.filter(
-                provider=provider,
-                client=client,
-                date__month=past.month if draft_invoice else now.month,
-            ).latest('date')
-            if create_invoice:
+            try:
+                invoice = models.Invoice.objects.filter(
+                    provider=provider,
+                    client=client,
+                    date__month=past.month if draft_invoice else now.month,
+                ).latest('date')
+            except models.Invoice.DoesNotExist:
+                # This may be the provider's first invoice.
+                invoice = None
+
+            if create_invoice or not invoice:
```

Comment:
@UmanShahzad Would it make sense to remove the `create_invoice` parameter from this function and default to always creating an invoice based on the latest data? That way the admin wouldn't need to remember to do this, and we'd make sure that the system wouldn't send a stale invoice.

Reply:
@itsjeyd But then this creates multiple invoices since we'd go through this twice (once for the approval, and once for finalization). This then messes with how we decided to use TransferWiseBulkPayments -- get all unpaid invoices for a certain date range (from ancient history to now by default).

Reply:
@UmanShahzad Ah, good point about bulk payments! Never mind, then. I don't think it would be impossible to get rid of `create_invoice` here and still deal with bulk payments properly. But I see now that this would require some additional conceptual changes to be able to select the right invoice if there's more than one invoice to choose from for a given provider and month. Fine to keep this as is :)